### ISSUE TITLE
504 Error when using an existing remote repo

### ISSUE THREAD

**Issue by Tavlor** [OpenMediaVault-Plugin-Developers/openmediavault-borgbackup#44]
I am trying to set up an existing remote repo (I succeeded, but I think by accident - more on that later).

When I try to save the config I get a 504 error (after a minute or so). The in-UI error message isn't very useful, but trying again gets a pretty instant error due to an unreleased lock (I released it from the command line). That error gave me the command that was run (including env vars I had set). From what I can tell, the pre-save config check times out because `borg info` takes too long.
- I used the [guide from the forum](https://forum.openmediavault.org/index.php?thread/36602-how-to-make-backups-with-borg-using-borgbackup-plugin/)
- I have `BORG_RSH` configured properly
- I tested `borg info` directly from the command line; it takes 2-3 minutes to finish (if I restrict it to the latest archive)
- I set `BORG_REPO` to `::latest` for repo creation, I don't know if it had an impact

I _was_ able to get the repo config to save - I saw the yellow "Pending configuration changes" popup after a bit, which applied what I had entered into the form. I think one of these happened:
- The background `borg info` command finished and the info was saved (triggering the popup)
- The popup happed when OMV tries to sign me out
- Working from another tab on the env variables triggered the popup, which captured the repo config too

I can now use the list button on the repo and get a result in seconds. The info button works too; I get the initial output quickly but the size estimate takes a while.

If you can give me some pointers on where to look, I'm happy to do some debugging

**Comment by ryecoaaron** [comment]
I don't think this is a plugin issue but I will setup something like the guide suggests to test.

**Comment by ryecoaaron** [comment]
I just tested the guide on a small remote repo.  It worked.  So, I think you are right and your setup is timing out because borg info is taking so long.

From the command line, if you do a borg info "path/to/repo" --last 1, is it any faster?  If not, is borg list "path/to/repo" fast?  The plugin needs something to confirm the remote repo is actually a borg repo.

**Comment by m-rios** [comment]
I have a similar issue. I can confirm that running `borg info` with `--last1` is a lot faster (~10s) `borg list` is also very fast (sub-second) for my repository with 40 archives

**Comment by ryecoaaron** [comment]
I have a remote repo running on an RPi4 with 41 archives.
All archives:              233.44 TB            227.57 TB             11.60 TB

I see the following times:

borg info - about 3.2 secs
borg list - about 2.7 secs
borg info --last 1 - about 13.6 secs
borg list --consider-checkpoints --short - about 2.7 secs

I'm thinking I will change it to the last command.
